plugins/zoom/icon_zoom/util/api.py

```diff
@@ -142,23 +192,31 @@ def _handle_response(self, response: Response, allow_404: bool, original_call_ar
             raise PluginException(preset=PluginException.Preset.BAD_REQUEST, data=response.json())
 
         if response.status_code == 401:
-            self.logger.info(f"Received HTTP {response.status_code}, re-authenticating to Zoom...")
-            self.refresh_oauth_token()
-            return self._call_api(**original_call_args)
+            if self.oauth_token or self._is_using_oauth():
+                self.logger.info(f"Received HTTP {response.status_code}, re-authenticating to Zoom...")
+                self._refresh_oauth_token()
+                return self._call_api(**original_call_args)
```

Comment:
I feel like we might need a call counter here 'cause we could end up in an infinite loop of trying to request a new token, getting a 401 back, and then trying again forever.